File: database/conexaoFrontBackEndBD.php
<?php
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Methods: GET, POST, OPTIONS");
header("Access-Control-Allow-Headers: Content-Type");
header("Content-Type: application/json; charset=UTF-8");

if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
    http_response_code(200);
    exit();
}

$DB_CONFIG = [
    "user" => "root",
    "password" => "",
    "host" => "localhost",
    "database" => "enchentes_BD"
];

function conectar_mysql($config) {
    $conn = new mysqli($config["host"], $config["user"], $config["password"], $config["database"]);
    if ($conn->connect_error) {
        http_response_code(500);
        echo json_encode(["erro" => "Erro na conexão: " . $conn->connect_error]);
        exit();
    }
    return $conn;
}

// Roteamento baseado em parâmetro GET 'rota'
$rota = $_GET['rota'] ?? '';

if ($rota == "cadastrar" && $_SERVER['REQUEST_METHOD'] === "POST") {
    cadastrar($DB_CONFIG);
} elseif ($rota == "api_ocorrencias") {
    api_ocorrencias($DB_CONFIG);
} elseif ($rota == "ocorrencias") {
    ocorrencias($DB_CONFIG);
} else {
    http_response_code(404);
    echo json_encode(["erro" => "Endpoint não encontrado"]);
    exit();
}

function cadastrar($config) {
    $cidade = $_POST['cidade'] ?? null;
    $nivel_agua = $_POST['nivel_agua'] ?? null;
    $pessoas_afetadas = $_POST['pessoas_afetadas'] ?? null;
    $data_enchente = $_POST['data_enchente'] ?? null;

    $nivel_agua = is_numeric($nivel_agua) ? floatval($nivel_agua) : null;
    $pessoas_afetadas = is_numeric($pessoas_afetadas) ? intval($pessoas_afetadas) : null;

    $conn = conectar_mysql($config);

    $query = "INSERT INTO registros (cidade, nivel_agua, pessoas_afetadas, data_enchente) VALUES (?, ?, ?, ?)";
    $stmt = $conn->prepare($query);
    $stmt->bind_param("sdss", $cidade, $nivel_agua, $pessoas_afetadas, $data_enchente);

    if ($stmt->execute()) {
        echo json_encode(["mensagem" => "Ocorrência cadastrada com sucesso!"]);
    } else {
        http_response_code(500);
        echo json_encode(["erro" => "Erro ao cadastrar ocorrência: " . $stmt->error]);
    }
    $stmt->close();
    $conn->close();
}

function api_ocorrencias($config) {
    $conn = conectar_mysql($config);
    $query = "SELECT * FROM registros";
    $result = $conn->query($query);

    $ocorrencias = [];
    while ($row = $result->fetch_assoc()) {
        $ocorrencias[] = $row;
    }
    echo json_encode($ocorrencias);

    $conn->close();
}

function ocorrencias($config) {
    api_ocorrencias($config);
}
?>

File: database/conexaoFrontBackEndBD.py
from flask import Flask, request, jsonify

from flask import Flask, render_template
import mysql.connector
from flask_cors import CORS

app = Flask(__name__)
CORS(app)

DB_CONFIG = {
    "user": "root", # coloca seu usuario do mysql, essa é a minha no caso
    "password": "root", # sua senha do mysql, essa é a minha no caso
    "host": "localhost",
    "database": "enchentes_BD"
}

def conectar_mysql():
    return mysql.connector.connect(**DB_CONFIG)

@app.route("/cadastrar", methods=["POST"])
def cadastrar():
    nome_local = request.form.get("nome_local")
    nivel_agua = request.form.get("nivel_agua")
    pessoas_afetadas = request.form.get("pessoas_afetadas")
    data_enchente = request.form.get("data_enchente")

    try:
        nivel_agua = float(nivel_agua) if nivel_agua else None
    except ValueError:
        nivel_agua = None

    try:
        pessoas_afetadas = int(pessoas_afetadas) if pessoas_afetadas else None
    except ValueError:
        pessoas_afetadas = None

    conn = conectar_mysql()
    cursor = conn.cursor()
    query = """
        INSERT INTO registros (nome_local, nivel_agua, pessoas_afetadas, data_enchente)
        VALUES (%s, %s, %s, %s)
    """
    cursor.execute(query, (nome_local, nivel_agua, pessoas_afetadas, data_enchente))
    conn.commit()
    cursor.close()
    conn.close()
    return jsonify({"mensagem": "Ocorrência cadastrada com sucesso!"})


@app.route("/ocorrencias")
def ocorrencias():
    conn = conectar_mysql()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("SELECT * FROM registros")
    ocorrencias = cursor.fetchall() 
    cursor.close()
    conn.close()
    return render_template("ocorrencias.html", ocorrencias=ocorrencias)

@app.route("/api/ocorrencias")
def api_ocorrencias():
    conn = conectar_mysql()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("SELECT * FROM registros")
    ocorrencias = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(ocorrencias)

if __name__ == "__main__":
    app.run(debug=True)